scipy/sparse/linalg/isolve/tfqmr.py

```diff
@@ -0,0 +1,182 @@
+import warnings
+import numpy as np
+from .utils import make_system
+
+
+__all__ = ['tfqmr']
+
+
+def tfqmr(A, b, x0=None, tol=1e-5, maxiter=None, M=None,
+          callback=None, atol=None, view=False):
+    """
+    Use Transpose-Free Quasi-Minimal Residual iteration to solve ``Ax = b``.
+
+    Parameters
+    ----------
+    A : {sparse matrix, dense matrix, LinearOperator}
+        The real or complex N-by-N matrix of the linear system.
+        Alternatively, `A` can be a linear operator which can
+        produce ``Ax`` using, e.g.,
+        `scipy.sparse.linalg.LinearOperator`.
+    b : {array, matrix}
+        Right hand side of the linear system. Has shape (N,) or (N,1).
+    x0  : {array, matrix}
+        Starting guess for the solution.
+    tol, atol : float, optional
+        Tolerances for convergence, ``norm(residual) <= max(tol*norm(b-Ax0), atol)``.
+        The default for `tol` is 1.0e-5
+        The default for `atol` is ``tol * norm(b-Ax0)``.
+
+        .. warning::
+
+           The default value for `atol` will be changed in a future release.
+           For future compatibility, specify `atol` explicitly.
+    maxiter : int, optional
+        Maximum number of iterations.  Iteration will stop after maxiter
+        steps even if the specified tolerance has not been achieved.
+    M : {sparse matrix, dense matrix, LinearOperator}
+        Inverse of the preconditioner of A.  M should approximate the
+        inverse of A and be easy to solve for (see Notes).  Effective
+        preconditioning dramatically improves the rate of convergence,
+        which implies that fewer iterations are needed to reach a given
+        error tolerance.  By default, no preconditioner is used.
+    callback : function, optional
+        User-supplied function to call after each iteration.  It is called
+        as `callback(xk)`, where `xk` is the current solution vector.
+    view : bool, optional
+        Specify ``view = True`` to view the convergence, ``view = False`` is
+        to close the output of the convergence.
+        Default is `False`
+
+    Returns
+    -------
+    x : array or matrix
+        The converged solution.
+    info : int
```

Comment:
The code returns `maxiter`?

Reply:
If the solution is converged, `info` equals zero. Otherwise, `info` equals the number of iterations (in this case, that is `info = maxiter`. The final iterative step reach to `maxiter` because linear solution is not converged).

Reply:
makes sense, thanks for clarifying